datetime/datetime_test.go

```diff
@@ -0,0 +1,543 @@
+	if err != nil {
+		t.Fatalf("Datetime delete failed: %s", err.Error())
+	}
+}
+
+type Event struct {
+	Datetime time.Time
+	Location string
+}
+
+type Tuple struct {
+	Cid    uint
+	Orig   string
+	Events []Event
+}
+
+// Same effect in a "magic" way, but slower.
+type Tuple2 struct {
+	_msgpack struct{} `msgpack:",asArray"` //nolint
+
+	Cid    uint
+	Orig   string
+	Events []Event
+}
+
+type Tuple3 struct {
+	Datetime time.Time
+}
+
+func (t *Tuple3) EncodeMsgpack(e *msgpack.Encoder) error {
+	if err := e.EncodeSliceLen(2); err != nil {
+		return err
+	}
+	if err := e.Encode(t.Datetime); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (t *Tuple3) DecodeMsgpack(d *msgpack.Decoder) error {
+	var err error
+	var l int
+	if l, err = d.DecodeSliceLen(); err != nil {
+		return err
+	}
+	if l != 1 {
+		return fmt.Errorf("array len doesn't match: %d", l)
+	}
+	err = d.Decode(&t.Datetime)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (ev *Event) EncodeMsgpack(e *msgpack.Encoder) error {
+	if err := e.EncodeSliceLen(2); err != nil {
+		return err
+	}
+	if err := e.EncodeString(ev.Location); err != nil {
+		return err
+	}
+	if err := e.Encode(ev.Datetime); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (ev *Event) DecodeMsgpack(d *msgpack.Decoder) error {
+	var err error
+	var l int
+	if l, err = d.DecodeSliceLen(); err != nil {
+		return err
+	}
+	if l != 2 {
+		return fmt.Errorf("array len doesn't match: %d", l)
+	}
+	if ev.Location, err = d.DecodeString(); err != nil {
+		return err
+	}
+	res, err := d.DecodeInterface()
+	if err != nil {
+		return err
+	}
+	ev.Datetime = res.(time.Time)
+	return nil
+}
+
+func (c *Tuple) EncodeMsgpack(e *msgpack.Encoder) error {
+	if err := e.EncodeSliceLen(3); err != nil {
+		return err
+	}
+	if err := e.EncodeUint(c.Cid); err != nil {
+		return err
+	}
+	if err := e.EncodeString(c.Orig); err != nil {
+		return err
+	}
+	e.Encode(c.Events)
+	return nil
+}
+
+func (c *Tuple) DecodeMsgpack(d *msgpack.Decoder) error {
+	var err error
+	var l int
+	if l, err = d.DecodeSliceLen(); err != nil {
+		return err
+	}
+	if l != 3 {
+		return fmt.Errorf("array len doesn't match: %d", l)
+	}
+	if c.Cid, err = d.DecodeUint(); err != nil {
+		return err
+	}
+	if c.Orig, err = d.DecodeString(); err != nil {
+		return err
+	}
+	if l, err = d.DecodeSliceLen(); err != nil {
+		return err
+	}
+	c.Events = make([]Event, l)
+	for i := 0; i < l; i++ {
+		d.Decode(&c.Events[i])
+	}
+	return nil
+}
+
+func TestCustomEncodeDecodeTuple1(t *testing.T) {
+	if isDatetimeSupported == false {
+		t.Skip("Skipping test for Tarantool without datetime support in msgpack")
+	}
+
+	conn := connectWithValidation(t)
+	defer conn.Close()
+
+	dt1, _ := time.Parse(time.RFC3339, "2010-05-24T17:51:56.000000009Z")
+	dt2, _ := time.Parse(time.RFC3339, "2022-05-24T17:51:56.000000009Z")
+
+	tuple := Tuple{Cid: 13, Orig: "orig", Events: []Event{{dt1, "Minsk"}, {dt2, "Moscow"}}}
+	resp, err := conn.Replace("testDatetime_3", &tuple)
```

Comment:
There is no check of the received datetime value. It seems to me that the test is not complete without the check.